tutorials/data_persistence/c++11/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+#
+# (c) 2024 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+#
+# RTI grants Licensee a license to use, modify, compile, and create derivative
+# works of the Software.  Licensee has the right to distribute object form
+# only for use with RTI products.  The Software is provided "as is", with no
+# warranty of any type, including any warranty for fitness for any purpose.
+# RTI is under no obligation to maintain or support the Software.  RTI shall
+# not be liable for any incidental or consequential damages arising out of the
+# use or inability to use the software.
+#
+
+cmake_minimum_required(VERSION 3.11)
+project(connext-tutorial-data-persistence)
+list(APPEND CMAKE_MODULE_PATH
+    "${CMAKE_CURRENT_SOURCE_DIR}/../../../resources/cmake/Modules"
+)
+include(ConnextDdsConfigureCmakeUtils)
+connextdds_configure_cmake_utils()
+
+# Include ConnextDdsAddExample.cmake from resources/cmake
+include(ConnextDdsAddExample)
+
+connextdds_call_codegen(
+    IDL "temperature"
+    LANG "C++11"
+    PREFIX "temperature"
+)
+
+connextdds_add_application(
+    TARGET "publisher"
+    LANG "C++11"
+    OUTPUT_NAME "temperature_publisher"
+    SOURCES
+        $<TARGET_OBJECTS:temperature_CXX11_obj>
+        "${CMAKE_CURRENT_SOURCE_DIR}/temperature_publisher.cxx"
+    DEPENDENCIES
+    ${_CONNEXT_DEPENDENCIES}
+)
+
+connextdds_add_application(
+    TARGET "subscriber"
+    LANG "C++11"
+    OUTPUT_NAME "temperature_subscriber"
+    SOURCES
+        $<TARGET_OBJECTS:temperature_CXX11_obj>
+        "${CMAKE_CURRENT_SOURCE_DIR}/temperature_subscriber.cxx"
+    DEPENDENCIES
+    ${_CONNEXT_DEPENDENCIES}
+)
+
+connextdds_add_application(
+    TARGET "durable_subscriber"
+    LANG "C++11"
+    OUTPUT_NAME "temperature_durable_subscriber"
+    SOURCES
+        $<TARGET_OBJECTS:temperature_CXX11_obj>
+        "${CMAKE_CURRENT_SOURCE_DIR}/temperature_durable_subscriber.cxx"
```

Comment:
Does temperature_durable_subscriber work when linked statically?

Reply:
Yes it does, and you don't need to have the libs in the `LD_LIBRARY_PATH`:
```
(.venv) ➜  build git:(feature/EVALG-49-data-persistence) ✗ cmake -DBUILD_SHARED_LIBS=OFF .. 
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- RTI Connext installation directory: /home/arobles/connext_installations/rti_connext_dds-7.3.0
-- RTI Connext architecture: x64Linux4gcc7.3.0
-- Found RTIConnextDDS: /home/arobles/connext_installations/rti_connext_dds-7.3.0 (found suitable version "7.3.0.0", minimum required is "6.1.0") found components: core 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/arobles/Desktop/GSG/rticonnextdds-examples/tutorials/data_persistence/c++11/build
(.venv) ➜  build git:(feature/EVALG-49-data-persistence) ✗ cd ..
(.venv) ➜  c++11 git:(feature/EVALG-49-data-persistence) ✗ cmake --build build                                                                                        
Scanning dependencies of target temperature_CXX11_sources
[  7%] Generating src/temperature.cxx, src/temperaturePlugin.cxx, src/temperature.hpp, src/temperaturePlugin.hpp, src/temperature_timestamp.cmake
INFO com.rti.ndds.nddsgen.Main Running rtiddsgen version 4.3.0, please wait ...
INFO com.rti.ndds.nddsgen.Main Done
[  7%] Built target temperature_CXX11_sources
Scanning dependencies of target temperature_CXX11_obj
[ 15%] Building CXX object CMakeFiles/temperature_CXX11_obj.dir/src/temperature.cxx.o
[ 23%] Building CXX object CMakeFiles/temperature_CXX11_obj.dir/src/temperaturePlugin.cxx.o
[ 30%] Built target temperature_CXX11_obj
Scanning dependencies of target cpp2_qos
[ 38%] Copying USER_QOS_PROFILES.xml
[ 38%] Built target cpp2_qos
Scanning dependencies of target durable_publisher_cpp2
[ 46%] Building CXX object CMakeFiles/durable_publisher_cpp2.dir/temperature_durable_publisher.cxx.o
[ 53%] Linking CXX executable temperature_durable_publisher
[ 53%] Built target durable_publisher_cpp2
Scanning dependencies of target durable_subscriber_cpp2
[ 61%] Building CXX object CMakeFiles/durable_subscriber_cpp2.dir/temperature_durable_subscriber.cxx.o
[ 69%] Linking CXX executable temperature_durable_subscriber
[ 69%] Built target durable_subscriber_cpp2
Scanning dependencies of target subscriber_cpp2
[ 76%] Building CXX object CMakeFiles/subscriber_cpp2.dir/temperature_subscriber.cxx.o
[ 84%] Linking CXX executable temperature_subscriber
[ 84%] Built target subscriber_cpp2
Scanning dependencies of target publisher_cpp2
[ 92%] Building CXX object CMakeFiles/publisher_cpp2.dir/temperature_publisher.cxx.o
[100%] Linking CXX executable temperature_publisher
[100%] Built target publisher_cpp2
(.venv) ➜  c++11 git:(feature/EVALG-49-data-persistence) ✗ ./build/temperature_durable_subscriber sub1
Sensor1: 34.00 degrees (1716967560.06s)
Sensor1: 30.00 degrees (1716967560.77s)
Sensor1: 34.00 degrees (1716967561.77s)
Sensor1: 36.00 degrees (1716967562.77s)
Sensor1: 39.00 degrees (1716967563.77s)
Sensor1: 30.00 degrees (1716967564.77s)
^C
(.venv) ➜  c++11 git:(feature/EVALG-49-data-persistence) ✗ ./build/temperature_durable_subscriber sub1
Sensor1: 33.00 degrees (1716967578.78s)
Sensor1: 36.00 degrees (1716967579.31s)
Sensor1: 37.00 degrees (1716967580.31s)
Sensor1: 39.00 degrees (1716967581.31s)
Sensor1: 32.00 degrees (1716967582.31s)
Sensor1: 31.00 degrees (1716967583.31s)
Sensor1: 32.00 degrees (1716967584.31s)
Sensor1: 34.00 degrees (1716967585.31s)
Sensor1: 35.00 degrees (1716967586.31s)
Sensor1: 33.00 degrees (1716967587.31s)
^C
```